### ISSUE TITLE
Difference in validation accuracy in model.evaluate_generator with Shuffle = True vs Shuffle = False

### ISSUE THREAD

**Issue by amittal-tcd** [keras-team/keras#14118]
**System information**  
- Have I written custom code (as opposed to using example directory):  Standard code and functions
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):  Google Colab standard config
- TensorFlow backend (yes / no):  Yes
- TensorFlow version:  2.2.0
- Keras version:  2.3.1
- Python version:  3.6.9
- CUDA/cuDNN version:  V10.1.243
- GPU model and memory:  Google Colab standard config

The model.evaluate_generator function gives a different accuracy with "Shuffle = True" vs "Shuffle = False". I have read other threads where I came across setting the same worker for processing on the same thread. I did that. But there still is a big difference in the results which seems unlikely given the model records 95% accuracy on training data and 93% accuracy on validation with shuffle = True.

I understand shuffling data in validation generators does not make sense, but I am should the result be different with shuffling in evaluate function ?

I am confused which is the correct result. The 2 different scenarios and there results are present in the Colab notebook as follows for reproduction of the issue -

[Image Classification Accuracy Changing Due to Shuffle in Data Generator](https://colab.research.google.com/drive/1ahX3RNenL9NMISHhZns727VrrK_TGVUs?usp=sharing)

The problem can be reproduced using the code in the colab notebook above. A few important details about the notebook are -
1. The model was trained on ~500k images
2. The process used early stopping callback with a test data of 200K images separate from the above images
3. The validation images in the google colab notebook above are separate from the training and testing images above

![Evaluate generator accuracy issue](https://user-images.githubusercontent.com/54919036/84573476-8ec30080-ad98-11ea-879d-9551aa0a9f52.JPG)
